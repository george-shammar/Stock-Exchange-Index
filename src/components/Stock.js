import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import '../Index.css';
/* eslint-disable */
const Stock = ({ stock, children }) => {

  return (
    <div key={stock.symbol} className="stock">
      <span className="stock-container">
        <h3 className="stock-name">{stock.symbol}</h3>
        <h4 className="stock-others">{stock.name}</h4>
        <h6 className="stock-price">{stock.price}</h6>
        <h5 className="stock-others">{stock.exchange}</h5>
        <Link to="/DetailPage">View Details</Link>
        <div className="image-container" />
      </span>
      {children}
    </div>
  );
};

Stock.propTypes = {
  children: PropTypes.instanceOf(Object).isRequired,
  stock: PropTypes.instanceOf(Object).isRequired,
};

export default Stock;
