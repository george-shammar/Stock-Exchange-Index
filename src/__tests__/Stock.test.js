import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Stock from '../components/Stock';

const stock = {
  symbol: 'Sym',
  price: '13.0',
  exchange: 'NASDAQ',
  name: 'Apple',
};

it('renders Stock without crashing', () => {
  const div = document.createElement('div');

  ReactDom.render(
    <BrowserRouter>
      <Stock stock={stock} key={stock.symbol} />
    </BrowserRouter>,
    div,
  );

  ReactDom.unmountComponentAtNode(div);
});

it('renders the Stock component correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Stock stock={stock} key={stock.symbol} />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
