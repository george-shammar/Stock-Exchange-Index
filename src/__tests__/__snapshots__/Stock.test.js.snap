// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NavBar component correctly 1`] = `
<div
  className="stock"
>
  <span
    className="stock-container"
  >
    <h3
      className="stock-name"
    >
      Sym
    </h3>
    <h4
      className="stock-others"
    >
      Apple
    </h4>
    <h6
      className="stock-price"
    >
      13.0
    </h6>
    <h5
      className="stock-others"
    >
      NASDAQ
    </h5>
    <a
      className="view mt-3"
      href="/DetailPage"
      onClick={[Function]}
    >
      <p>
        View Details
      </p>
    </a>
  </span>
</div>
`;

exports[`renders the Stock component correctly 1`] = `
<div
  className="stock"
>
  <span
    className="stock-container"
  >
    <h3
      className="stock-name"
    >
      Sym
    </h3>
    <h4
      className="stock-others"
    >
      Apple
    </h4>
    <h6
      className="stock-price"
    >
      13.0
    </h6>
    <h5
      className="stock-others"
    >
      NASDAQ
    </h5>
    <a
      className="view mt-3"
      href="/DetailPage"
      onClick={[Function]}
    >
      <p>
        View Details
      </p>
    </a>
  </span>
</div>
`;
